### ISSUE TITLE
No error in analyzer for the mixin declaration on two same classes

### ISSUE THREAD

**Issue by sgrekhov** [dart-lang/sdk#45505]
The following code produces an error in CFE and no issues in analyzer

```dart
class C {}
typedef CAlias = C;

mixin CM on C, CAlias {}

main() {
}
```
CFE output is 
```
Error: 'C' can only be implemented once.
Error: Compilation failed.
```
Analyzer output is `No issues found!`
`Dart SDK version: 2.13.0-165.0.dev (dev) (Thu Mar 25 02:53:52 2021 -0700) on "windows_x64"`

**Comment by eernstg** [comment]
This is actually already mentioned in https://github.com/dart-lang/sdk/issues/45489#issuecomment-808227822. I added a reference to this issue there, to make sure that this issue is processed even in the case where the fault is resolved as part of #45489.

**Comment by devoncarew** [comment]
Thanks - I'm going to close this issue as a dup of https://github.com/dart-lang/sdk/issues/45489.
